Sources/XCHammer/XcodeTarget.swift

```diff
@@ -1262,6 +1272,70 @@ public class XcodeTarget: Hashable, Equatable {
 
         // Minimal settings for this build
         var settings = XCBuildSettings()
+
+        let xcodeTarget = self
+        /// We need to include the sources into the target
+        let sources: [ProjectSpec.TargetSource]
+        let xcodeBuildableTargetSettings: XCBuildSettings
+        let deps: [ProjectSpec.Dependency]
+
+        let pathsPredicate = makePathFiltersPredicate(genOptions.pathsSet)
+        let linkedDeps = xcodeTarget.linkedTargetLabels
+            .compactMap { targetMap.xcodeTarget(buildLabel: $0, depender: xcodeTarget) }
+            .filter { includeTarget($0, pathPredicate: pathsPredicate) }
+            .map { ProjectSpec.Dependency(type: .target, reference: $0.xcTargetName + "-Bazel",
+                    embed: $0.isExtension) }
+
+        if isTopLevelTestTarget {
+            let flattened = Set(flattenedInner(targetMap: targetMap))
+            // Determine deps to fuse into the rule.
+            let pathsPredicate = makePathFiltersPredicate(genOptions.pathsSet)
+            let fusableDeps = self.unfilteredDependencies
+                .filter { flattened.contains($0) && includeTarget($0, pathPredicate:
+                        pathsPredicate) }
+
+            xcodeBuildableTargetSettings = self.settings
+                            <> fusableDeps.foldMap { $0.settings }
+
+            if let xcBuildTestHost = xcodeBuildableTargetSettings.testHost?.v {
+                 // Notes on test host build configuration:
+                 // - Xcode bazel-builds the app as a scheme dep
+                 // - Xcode bazel-builds the test which install the test bundle into the
+                 //   app
+                 settings.testHost = First(xcBuildTestHost.replacingOccurrences(of: ".app", with: "-Bazel.app") + "-Bazel")
+            }
+
+
+            // Use settings, sources, and deps from the fusable deps
+            sources = fusableDeps.flatMap { $0.xcCompileableSources }
+
+            if shouldPropagateDeps(forTarget: xcodeTarget) {
+                deps = fusableDeps
+                    .flatMap { $0.transitiveDeps } + xcodeTarget.xcExtensionDeps
+            } else {
+                deps = []
+            }
+
+
+            // We need to stub out the CC
+        } else {
+            sources = self.xcCompileableSources
+            if shouldPropagateDeps(forTarget: xcodeTarget) {
+                deps = Array(xcodeTarget.transitiveDeps) +
+                    xcodeTarget.xcExtensionDeps
+            } else {
+                deps = []
+            }
+            xcodeBuildableTargetSettings = self.settings
+            settings.infoPlistFile = xcodeBuildableTargetSettings.infoPlistFile
+        }
+
+        settings.cc = First("$(PROJECT_FILE_PATH)/XCHammerAssets/xcode_clang_stub.sh")
```

Comment:
Longer term, I don't think we'll have to stub out clang and ld once we replace xcbuild. This is a stopgap and a means to an end to have a way to run tests without replacing xcbuild